Documentation/network/concepts/masquerading.rst

```diff
@@ -44,6 +44,17 @@ Setting the routable CIDR
 Setting the masquerading interface
   See :ref:`masq_modes` for configuring the masquerading interfaces.
 
+Masquerade traffic to Remote Nodes
+**********************************
+
+To masquerade traffic to remote nodes in BPF masquerading mode, use the option ``enable-remote-node-masquerade: "true"``. 
+This option requires ``enable-bpf-masquerade: "true"`` and also either 
+``enable-ipv4-masquerade: "true"`` or ``enable-ipv6-masquerade: "true"`` to SNAT traffic for IPv4 and IPv6, respectively.
+
+However, this flag does not provide the optimal behavior because packets directed to the nodes' InternalIP addresses will also be masqueraded. 
+To achieve the optimal behavior, we would need to inform the datapath about the type of IP address each node IP address represents. 
+This could potentially be accomplished by extending the node ID map.
```

Comment:
This is user documentation describing the concepts, but the text appears to be proposing future changes that may or may not be put into Cilium. I would suggest to make the text clear and specific about the current behavior and leave forward-looking statements for issue discussions.
```suggestion
This flag currently masquerades traffic to node ``InternalIP`` addresses. This may change in future. See :gh-issue:`####` for further discussion on this topic.
```

Note the suggestion above does not specify a specific github issue; you might consider filing a new one for the proposed solution, or alternatively you could reuse one of the existing ones. Depends whether we intend to close those existing issues assuming this PR is accepted.

Reply:
Done in the new commit